react-components/src/hooks/use3DModelName.tsx

```diff
@@ -0,0 +1,22 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+import { useSDK } from '../components/RevealContainer/SDKProvider';
+
+export const use3DModelName = (ids: number[]): UseQueryResult<string[] | undefined, unknown> => {
+  const sdk = useSDK();
+
+  const queryResult = useQuery<string[] | undefined>(['cdf', '3d', 'model', ids], async () => {
+    const modelNames: string[] = await Promise.all(
```

Comment:
Suggest using Promise.allSettled instead, Promise.all will throw an error if one of the requests fails, while Promise.allSettled will resolve all the requests and have a status with the resolved request. That way you won't "crash the app" if one of the models is missing. 



```suggestion
    const modelNames: string[] = await Promise.allSettled(
```